### ISSUE TITLE
Textures of type OF_PIXELS_YUY2 aren't drawn correctly at custom sizes.

### ISSUE THREAD

**Issue by bakercp** [openframeworks/openFrameworks#4990]
It seems that the programmable renderer is having some problems rendering YUY2 video textures when asked to do it at widths and heights other than the native video width / height.

For example:

```cpp
        g = std::make_shared<ofVideoGrabber>();

        // Use the ofxPS3EyeGrabber.
        grabber->setGrabber(std::make_shared<ofxPS3EyeGrabber>());

        // The native pixel format for the ofxPS3EyeGrabber is OF_PIXELS_YUY2
        // (aka YUV422).  When used this way, no additional pixel copies are made
        // or colorspace conversions are performed.
        //
        // The programmable renderer is able to directly render YUV422 pixels.
        // so be sure to that the OpenGL version is > 3.2, otherwise you'll
        // get a blank screen.
        grabber->setPixelFormat(OF_PIXELS_NATIVE); // This means the internal pixel type is OF_PIXELS_YUY2.
        grabber->setup(camWidth, camHeight);
    }
```

Then we draw
```cpp
        float w = g->getWidth();
        float h = g->getHeight();

        g->draw(0, 0, w, h);
        g->draw(w, 0, w / 2, h / 2);
        g->draw(0, h, w / 1.5, h / 1.5);
```

This produces the following output:

![screenshot_3_13_16__6_36_pm](https://cloud.githubusercontent.com/assets/300484/13732226/8b6b8760-e94b-11e5-8077-0ba5ae9f8e2a.png)

So it looks like the internal shaders are correctly drawing the YUY2 planes when the passed video / texture size is the one that is set up in the video player ... but when the size is manually changed (e.g. via draw(), then something gets messed up when drawing these YUY2 textures.

I believe it is happening because when binding the `ofBaseVideoDraws`, the shader uniforms aren't made aware of the requested non-native video size, leaving the shader to mis-align YUY2 planes when executing the fragment shader.

My investigation is paused here for the moment because I think @arturoc probably has the best angle on this problem :)

**Comment by arturoc** [comment]
I think this is just an artifact from sampling the U and V planes with bilinear filtering. i don't think there's an easy solution. i would try:

- using nearest neighbour filtering for the texture
- a possible solution is to round the texture coordinate to the next/previous valid pixel and then sample using that which will give a really weird resizing but the colours would be better. 
- additionally you can then sample the neighbouring pixels using the remainder and do the sampling manually

i don't think any of these except perhaps the latest would really solve the issue 100% but it would complicate the shader to much and even be slower so i don't think it's worth

the simplest solution is to render to an fbo and then draw the final rgb texture. this is something the grabber could even do internally (perhaps optionally) if there's several planes, so then if you ask for the texture with getTexture() what you get is an rgb texture which is much easier to work with. This could even happen in the video thread using a different GL context.

**Comment by bakercp** [comment]
Yeah, I was considering the internal FBO approach.  I'm not sure if there is a big enough speed boost associated with using YUY2 pixels rather than just doing a quick YUY2 -> RBG conversion, which of course solves all of this.  Since this camera in particular comes through RAM anyway, it may not be a huge speed loss.  Anyway ... will try things out.

**Comment by Teatoller** [comment]
Hi @bakercp,
I have followed the dialogue you had with @arturoc  on the subject quoted below.
Were you persuaded that the matter was best left as is?
If so, should issue #4990 be considered as resolved?

Regards
Steven Ennis.

> It seems that the programmable renderer is having some problems rendering YUY2 video textures when asked to do it at widths and heights other than the native video width / height.
> 
> For example:
> 
> ```c++
>         g = std::make_shared<ofVideoGrabber>();
> 
>         // Use the ofxPS3EyeGrabber.
>         grabber->setGrabber(std::make_shared<ofxPS3EyeGrabber>());
> 
>         // The native pixel format for the ofxPS3EyeGrabber is OF_PIXELS_YUY2
>         // (aka YUV422).  When used this way, no additional pixel copies are made
>         // or colorspace conversions are performed.
>         //
>         // The programmable renderer is able to directly render YUV422 pixels.
>         // so be sure to that the OpenGL version is > 3.2, otherwise you'll
>         // get a blank screen.
>         grabber->setPixelFormat(OF_PIXELS_NATIVE); // This means the internal pixel type is OF_PIXELS_YUY2.
>         grabber->setup(camWidth, camHeight);
>     }
> ```
> 
> Then we draw
> 
> ```c++
>         float w = g->getWidth();
>         float h = g->getHeight();
> 
>         g->draw(0, 0, w, h);
>         g->draw(w, 0, w / 2, h / 2);
>         g->draw(0, h, w / 1.5, h / 1.5);
> ```
> 
> This produces the following output:
> 
> ![screenshot_3_13_16__6_36_pm](https://cloud.githubusercontent.com/assets/300484/13732226/8b6b8760-e94b-11e5-8077-0ba5ae9f8e2a.png)
> 
> So it looks like the internal shaders are correctly drawing the YUY2 planes when the passed video / texture size is the one that is set up in the video player ... but when the size is manually changed (e.g. via draw(), then something gets messed up when drawing these YUY2 textures.
> 
> I believe it is happening because when binding the `ofBaseVideoDraws`, the shader uniforms aren't made aware of the requested non-native video size, leaving the shader to mis-align YUY2 planes when executing the fragment shader.
> 
> My investigation is paused here for the moment because I think @arturoc probably has the best angle on this problem :)
